src/app/parcels/ActionBar/Actions.tsx

```diff
@@ -252,6 +264,10 @@ const Actions: React.FC<Props> = ({
                             return;
                         case "Generate Map":
                             openInNewTab(mapsLinkForSelectedParcels());
+                            onActionCompleted(
+                                selectedParcels,
+                                availableActions["Generate Map"].newStatus
+                            );
```

Comment:
Do we really need this? Aren't we calling this twice for map generation effectively?

Reply:
The map generation action doesn't open an actions modal, unlike the other actions

Reply:
Oh i see! The way it's split up isn't ideal but i guess this will have to do for now

Reply:
Yep :(